File: apps/konekto_portal_next/lib/browserNotifications.ts
/**
 * Portado de apps/konekto_portal/lib/features/dashboard/browser_notifications.dart
 * — fina camada sobre a Notification API do navegador, sem service worker
 * (não é Web Push de verdade: só funciona com o portal aberto). Falha
 * silenciosamente se o navegador não suportar ou a permissão for negada.
 */
export const browserNotifications = {
  async requestPermissionIfNeeded(): Promise<void> {
    try {
      if (typeof Notification === 'undefined') return
      if (Notification.permission === 'default') {
        await Notification.requestPermission()
      }
    } catch {
      // Navegador sem suporte à Notification API — segue sem notificar.
    }
  },

  show({ title, body }: { title: string; body: string }): void {
    try {
      if (typeof Notification === 'undefined') return
      if (Notification.permission !== 'granted') return
      new Notification(title, { body, icon: '/icons/Icon-192.png' })
    } catch {
      // Ignora falhas de plataforma (ex: iOS Safari sem suporte).
    }
  },
}

File: apps/konekto_portal/lib/features/dashboard/browser_notifications.dart
import 'dart:js_interop';

import 'package:web/web.dart' as web;

/// Fininha camada sobre a Notification API do navegador — permissão e
/// exibição, sem service worker (não é Web Push de verdade: só funciona
/// enquanto o portal está aberto no navegador, mas não exige a
/// infraestrutura de push). Falha silenciosamente se o navegador não
/// suportar ou a permissão for negada — o alerta sonoro/visual dentro do
/// app continua funcionando de qualquer forma.
class BrowserNotifications {
  static Future<void> requestPermissionIfNeeded() async {
    try {
      if (web.Notification.permission == 'default') {
        await web.Notification.requestPermission().toDart;
      }
    } catch (_) {
      // Navegador sem suporte à Notification API — segue sem notificar.
    }
  }

  static void show({required String title, required String body}) {
    try {
      if (web.Notification.permission != 'granted') return;
      web.Notification(title, web.NotificationOptions(body: body, icon: 'icons/Icon-192.png'));
    } catch (_) {
      // Ignora falhas de plataforma (ex: iOS Safari sem suporte).
    }
  }
}
